docs/deployment.md

```diff
@@ -0,0 +1,25 @@
+# Remote Deployment
+
+This document will cover the process of deploying Wikijump on Amazon Web Services.
+
+## General Architecture
+
+Communication with Wikijump begins at DNS. If the address is for a wikijump.com subdomain, or a custom domain CNAMEd to wikijump.com, it will hit an Elastic Load Balancer. Specifically, it will resolve to the static (elastic) IP address of a Network Load Balancer listening on ports 80 and 443, but not terminating any SSL connections. It will proxy all this traffik to the Traefik edge proxy containers running in Elastic Container Service. Traefik handles the business of SSL termination and the permanent redirecting of http to https. It will then act as a reverse proxy, forwarding the request on to other containers running php-fpm and nginx. These containers are connected to a caching layer (memcached) and a database (postgres).
```

Comment:
An Elastic IP in AWS parlance is a public IP address that can be freely associated with a customer's assets, such as load balancers and EC2s. The IP is reserved by the account until explicitly released back to the pool so it functions as a static IP for a service. Without the usage of Elastic IP, every time the load balancer was recreated it would receive a new IP from the pool.